Microsoft/WebTemplateStudio
# re-architecture: extension call corets-cli

closes #955

**Summary to this change:**

1. put corets cli into build file, so run ./build, it builds the cli executable 
2. use a child process to start the corets cli executable 
3. send commands to cli and receive responses
4. interpret responses and return to UI
5. clean up/remove the code related to corets-api, including all the dlls

There are 6 commands in total:
1. sync command
2. get frameworks
3. get pages
4. generate

5. get features,it returns available Azure services based on the front-end and back-end frameworks ( not used in WebTS atm)
6. get project types ( not used in WebTS atm )


**How to test:**
1. switch to my branch and do git submodule update
2. do ./build
3. start the extension, and generate a project (with and without Azure services) you should be able to generate a project 
4. in post generation modal, click "create a new project", you should be able to generate another project
5. in post generation modal, click "close wizard" and do "ctrl+shift+p" to start a new wizard, you should be able to generate another project.

**Make sure to run:**
git submodule init
git submodule update
___

**Test Cases:**
- [ ] you are able to generate a project with Azure Services 
- [ ] you are able to generate a project without Azure Services 
- [ ] in post generation modal, click "close wizard" and do "ctrl+shift+p" to start a new wizard, you should be able to generate another project.
- [ ] in post generation modal, click "create a new project", you should be able to generate another project
